src/datasets/utils/streaming_download_manager.py

```diff
@@ -56,6 +56,14 @@ def xjoin(a, *p):
     return "::".join([a] + b)
 
 
+def _as_posix(path):
+    return path.as_posix().replace(":/", "://")
```

Comment:
I think it can cause an error in a case like this:
```
PosixPath('zip:/test.txt::/Users/username/bar.zip') -> 'zip://test.txt:://Users/username/bar.zip'
```
a simple workaround would be:
```suggestion
    return path.as_posix().replace(":/", "://").replace(":://", "::/")
```

Reply:
or, this
```suggestion
    return re.sub("([A-z]:/)([A-z])", "\g<1>/\g<2>", path.as_posix())
```

Reply:
Good catch!

Reply:
I finally did:
```python
return re.sub(r"(?<!:):/", "://", path.as_posix())
```